tests/ConversationTranslatorTests.ts

```diff
@@ -84,6 +86,401 @@ describe("conversation constructor tests", () => {
+
+        const c = sdk.Conversation.createConversationAsync(config, (() => {
+            expect(c.conversationId.length).toEqual(5);
+        }));
+
+        objsToClose.push(c);
+
+        WaitForCondition(() => (c.conversationId !== "" && c.conversationId.length === 5), done);
+
+    }, 20000);
+
+    test("Start Conversation, invalid language [400003]", (done: jest.DoneCallback) => {
+
+        let errorMessage: string;
+        const config = sdk.SpeechTranslationConfig.fromSubscription(Settings.SpeechSubscriptionKey, Settings.SpeechRegion);
+        config.speechRecognitionLanguage = "abc";
+        if (endpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Host], endpointHost); }
+        if (speechEndpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+        const c = sdk.Conversation.createConversationAsync(config, undefined, ((error: any) => {
+            expect(error).toContain("400003");
+            errorMessage = error;
+        }));
+        objsToClose.push(c);
+
+        WaitForCondition(() => (errorMessage !== undefined), done);
+
+    }, 20000);
+
+    test("Start Conversation, invalid nickname [400025]", (done: jest.DoneCallback) => {
+
+        let errorMessage: string;
+
+        const config = sdk.SpeechTranslationConfig.fromSubscription(Settings.SpeechSubscriptionKey, Settings.SpeechRegion);
+        config.speechRecognitionLanguage = "en-US";
+        if (endpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Host], endpointHost); }
+        if (speechEndpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+        config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Name], "Host.Testing");
+
+        const c = sdk.Conversation.createConversationAsync(config, undefined, ((error: any) => {
+            expect(error).toContain("400025");
+            errorMessage = error;
+        }));
+        objsToClose.push(c);
+
+        WaitForCondition(() => (errorMessage !== undefined), done);
+
+    }, 20000);
+
+    test("Start Conversation, invalid subscription key or region [401000]", (done: jest.DoneCallback) => {
+
+        let errorMessage: string;
+
+        const config = sdk.SpeechTranslationConfig.fromSubscription("abc", "def");
+        config.speechRecognitionLanguage = "en-US";
+        if (endpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Host], endpointHost); }
+        if (speechEndpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+        const c = sdk.Conversation.createConversationAsync(config, undefined, ((error: any) => {
+            expect(error).toContain("401000");
```

Comment:
>expect(error).toContain("401000"); [](start = 12, length = 34)

expect checks need to be wrapped in a try{expect}catch(e){done.fail(e)} or they'll get missed if they're in a callback.

Reply:
Expect checks are now wrapped in the callbacks